### PR TITLE
[no-jira] Add TimeZone field

### PR DESCRIPTION
**Pull request by knutsenm**
Help Scout 655225

### CODE REVIEW

**Comment by knutsenm** [comment]
Works in staging!

File: db/migrate/20211112184057_add_time_zone_field.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      option_value = OptionValue.find_or_create_by(label: label, name: name)
+      FieldOption.find_or_create_by(field: timezone, option_value: option_value, position: index)
+    end
+  end
+
+  def down
+    timezone = Field.find_by(name: "TimeZone")
+    timezone.option_values.each { |ov| ov.destroy } # destroy_all does not work
+    timezone.destroy
+  end
```
Comment (dbenton9) [inline db/migrate/20211112184057_add_time_zone_field.rb:19]: To make this migration reversible with `def down` you undo the Fields you created by destroying them here. Should you also destroy `OptionValue`'s and `FieldOption`'s created on lines 9-10?

Comment (knutsenm) [reply db/migrate/20211112184057_add_time_zone_field.rb:19]: Line 17 destroys the OptionValues. Line 18 destroys the FieldOptions as a side effect of `timezone.destroy` (`has_many :field_options, dependent: :destroy`)

Comment (knutsenm) [reply db/migrate/20211112184057_add_time_zone_field.rb:19]: Actually, line 17 probably destroys the FieldOptions first. OptionValue also `has_many :field_options, dependent: :destroy`

In any event, I tested this locally and it does work.

Comment (dbenton9) [reply db/migrate/20211112184057_add_time_zone_field.rb:19]: Sweet, if you tested the migration down this should be fine then 😃

**Review by andrewroth — COMMENTED** [review]
LGTM

**Pull request merged**